### PR TITLE
add option to log to stdout

### PR DESCRIPTION
**Pull request by ghost**
https://devtopia.esri.com/runtime/home-one/issues/2371

Add an option to log test output to stdout. This is easier than reading the results.zip and looking through individual log files for each test. The option is turned on by default

- [x] vTest (3rd): https://runtime-rtc.esri.com/view/vTest/job/vtest/job/3rdparty-interface/376/downstreambuildview/
- [x] vTest (rtc): https://runtime-rtc.esri.com/view/vTest/job/vtest/job/runtimecore-interface/8654/downstreambuildview/

### CODE REVIEW

**Review by colingpowell — COMMENTED** [review]
If we're running multiple tests in parallel, will the output from the various test tasks be mixed up in stdout?

**Comment by ghost** [comment]
> If we're running multiple tests in parallel, will the output from the various test tasks be mixed up in stdout?

It's not mixed up from what I see https://runtime-rtc.esri.com/view/vTest/job/vtest/job/runtimecore-linux/8841/consoleFull
Looking closer I don't see what's running the tests in parallel. I will take a little more time to understand it.

**Comment by colingpowell** [comment]
10-4, yeah, I don't think that's necessarily a deal-breaker, but something we'd have to keep in mind when looking at test output.

File: gtest_parallel.py [hunk 1]
```diff
@@ -1029,6 +1032,10 @@ def default_options_parser():
   parser.add_option('--serialize_test_cases', action='store_true',
                     default=False, help='Do not run tests from the same test '
                                         'case in parallel.')
+  parser.add_option('--log_to_stdout',
+                    choices=['yes', 'no'],
+                    default='yes',
+                    help='Print individual test output to stdout')
```
Comment (parkske) [inline gtest_parallel.py]: Do we need the choices, or should this just be enabled by passing `--log_to_stdout`?

**Review by john4744 — APPROVED** [review]

File: gtest_parallel.py (hunk 1 above)
Comment (ghost) [reply gtest_parallel.py]: In scrum discussion we decided to make it enabled by default. I'm not sure how to allow an option to turn it off without the choices.

Comment (parkske) [reply gtest_parallel.py]: Ah ok, fair point. This is fine then.

**Comment by ghost** [comment]
> > > If we're running multiple tests in parallel, will the output from the various test tasks be mixed up in stdout?

> > It's not mixed up from what I see https://runtime-rtc.esri.com/view/vTest/job/vtest/job/runtimecore-linux/8841/consoleFull
Looking closer I don't see what's running the tests in parallel. I will take a little more time to understand it.

> 10-4, yeah, I don't think that's necessarily a deal-breaker, but something we'd have to keep in mind when looking at test output.

I have a guess but am not sure. The tests are each run in separate parallel subprocesses but the subprocess output is captured into the main script process. We write that to a file. Then the call `sys.stdout.write(log_file.read())` effectively serializes output to stdout. I think that is either because of the file access or that `write` is already running while `read` is called.

**Comment by colingpowell** [comment]
> I have a guess but am not sure. The tests are each run in separate parallel subprocesses but the subprocess output is captured into the main script process. We write that to a file. Then the call sys.stdout.write(log_file.read()) effectively serializes output to stdout. I think that is either because of the file access or that write is already running while read is called.

Ah, right - that makes sense.

**Review by colingpowell — APPROVED** [review]

**Review by Craig-Proctor — APPROVED** [review]

File: gtest_parallel.py (hunk 1 above)
Comment (cfrankmiller) [reply gtest_parallel.py]: How about a `--quite` option?

Comment (ghost) [reply gtest_parallel.py]: Oh, that would work too. Maybe `--suppress_individual_test_output`, as there is other output which would not be turned off

Comment (cfrankmiller) [reply gtest_parallel.py]: Any of the suggested options are fine with me. Your call.

**Review by cfrankmiller — APPROVED** [review]

**Pull request merged**